lib/archethic/contracts.ex

```diff
@@ -165,6 +168,44 @@ defmodule Archethic.Contracts do
     end
   end
 
+  @doc """
+  Simulate the execution of the contract hold in prev_tx with the inputs of next_tx, at a certain date
+  """
+
+  @spec simulate_contract_execution!(Transaction.t(), Transaction.t(), DateTime.t()) ::
+          %{valid: false, address: binary(), reason: binary()} | %{valid: true, address: binary()}
+  def simulate_contract_execution!(
+        prev_tx = %Transaction{data: %TransactionData{code: code}},
+        next_tx = %Transaction{},
+        date = %DateTime{}
+      ) do
+    case Interpreter.parse(code) do
+      {:ok,
+       %Contract{
+         triggers: triggers,
+         conditions: %{inherit: transaction_conditions}
+       }} ->
+        constants = %{
+          "previous" => Constants.from_transaction(prev_tx),
+          "next" => Constants.from_transaction(next_tx)
+        }
+
+        with :ok <- validate_conditions(transaction_conditions, constants),
+             :ok <- validate_triggers(triggers, next_tx, date) do
+          ## The next command is guaranteed to suceed, we run the contract anyway to
+          ## maybe trigger possible exit from the process
+          ActionInterpreter.execute(Map.fetch!(triggers, :transaction), constants)
+          %{valid: true, address: prev_tx.address}
+        else
+          {:error, reason} ->
+            %{valid: false, address: prev_tx.address, reason: reason}
+        end
+
+      {:error, reason} ->
+        %{valid: false, address: prev_tx.address, reason: reason}
+    end
```

Comment:
I think this low level function should return a `{:ok, _} | {:error, _}` or even a `:valid | {:invalid, reason}`. This `%{valid: bool(), ...}` struct should only live in the controller. What do you think?